liquidity_pool/src/lib.rs

```diff
@@ -209,9 +209,9 @@ impl LiquidityPoolTrait for LiquidityPool {
         let total_shares = get_total_shares(&e);
 
         let zero = BigInt::zero(&e);
-        let new_total_shares = if reserve_a > zero.clone() && reserve_b > zero {
-            let shares_a = (balance_a.clone() * total_shares.clone()) / reserve_a;
-            let shares_b = (balance_b.clone() * total_shares.clone()) / reserve_b;
+        let new_total_shares = if reserve_a > zero && reserve_b > zero {
```

Comment:
This line returns the updated `total_shares`, not the delta being minted, which is why line 220 mints         `new_total_shares - total_shares`. Our test coverage here is weak though because this change still passes.

Reply:
[Uniswap V2](https://github.com/Uniswap/v2-core/blob/ee547b17853e71ed4e0101ccfd52e70d5acded58/contracts/UniswapV2Pair.sol#L111-L126) does balance - reserve in it's mint calculation, which may be what you're comparing against?

Reply:
You're right, not sure what I was thinking.